be/src/storage/tablet_updates.cpp

```diff
@@ -349,6 +350,7 @@ void TabletUpdates::_sync_apply_version_idx(const EditVersion& v) {
 
 void TabletUpdates::_redo_edit_version_log(const EditVersionMetaPB& v) {
     EditVersionInfo* tmp = new EditVersionInfo();
+    ScopedCleanup free_guard([&]() { delete tmp; });
```

Comment:
I think you should delete the pointers in _versions in the destructor of TabletUpdates, not add a cleanup here.

Reply:
free_guard.cancel();  
 this operation will cancel it, and the pointer will be store in _versions and delete be destructor of TabletUpdates

Reply:
I think It will be more clearr to use a `std::unique_ptr` here and release the pointer back into _versions at the last.
```
auto tmp = std::make_unique<EditVersionInfo>();
...
_versions.emplace_back(tmp.release());
```

Reply:
I see, `_versions.emplace_back` may throw an exception, and if so, the memory pointed by `tmp.release()` would be leaked. Thanks for pointing it out @laotan332.
Maybe we can define `_versions` as `std::vector<std::unique_ptr<EditVersionInfo>>`.